backend/elasticsearch/src/main/java/com/rbmhtechnology/vind/elasticsearch/backend/ElasticSearchServer.java

```diff
@@ -212,8 +212,8 @@ public SearchResult execute(FulltextSearch search, DocumentFactory factory) {
                 //TODO: if nested doc search is implemented
                 //final Map<String,Integer> childCounts = SolrUtils.getChildCounts(response);
 
-                final List<Document> documents =Arrays.stream(response.getHits().getHits())
-                        .map(hit -> DocumentUtil.buildVindDoc(hit.getSourceAsMap(), factory, search.getSearchContext()))
```

Comment:
Possible NPE?

Reply:
SearchHits is an object with an array of hits which is set to empty as default, but not null. It should never happen to have  NPI (if there is no change in elasticsearch response), nevertheless I'll add the check to avoid future surprises.